design_introspection-plugin/Makefile

```diff
@@ -0,0 +1,26 @@
+CXX = $(shell yosys-config --cxx)
+CXXFLAGS = $(shell yosys-config --cxxflags)
+LDFLAGS = $(shell yosys-config --ldflags)
```

Comment:
Can we template this Makefile?  I keep seeing it over and over?

Reply:
See PRs #43 and #44